public/app/features/alerting/unified/hooks/useCombinedRule.ts

```diff
@@ -247,38 +247,145 @@ export function useCombinedRule({ ruleIdentifier }: { ruleIdentifier: RuleIdenti
       }
     }
 
-    if (!dsSettings) {
+    return;
+  }, [ruleIdentifier, ruleSourceName, promRuleNs, rulerRuleGroup, ruleSource]);
+
+  return {
+    loading: isLoadingDsFeatures || isLoadingPromRules || isLoadingRulerGroup || rulerRuleGroupUninitialized,
+    error: ruleLocationError ?? promRuleNsError ?? rulerRuleGroupError,
+    result: rule,
+  };
+}
+
+interface RuleLocation {
+  namespace: string;
+  group: string;
+  ruleName: string;
+}
+
+function useRuleLocation(ruleIdentifier: RuleIdentifier): RequestState<RuleLocation> {
+  const { isLoading, currentData, error, isUninitialized } = alertRuleApi.endpoints.getAlertRule.useQuery(
+    { uid: isGrafanaRuleIdentifier(ruleIdentifier) ? ruleIdentifier.uid : '' },
+    { skip: !isGrafanaRuleIdentifier(ruleIdentifier), refetchOnMountOrArgChange: true }
+  );
+
+  return useMemo(() => {
+    if (isPrometheusRuleIdentifier(ruleIdentifier) || isCloudRuleIdentifier(ruleIdentifier)) {
+      return {
+        result: {
+          namespace: ruleIdentifier.namespace,
+          group: ruleIdentifier.groupName,
+          ruleName: ruleIdentifier.ruleName,
+        },
+        loading: false,
+      };
+    }
+
+    if (isGrafanaRuleIdentifier(ruleIdentifier)) {
+      if (isLoading || isUninitialized) {
+        return { loading: true };
+      }
+
+      if (error) {
+        return { loading: false, error };
+      }
+      if (currentData) {
+        return {
+          result: {
+            namespace: currentData.grafana_alert.namespace_uid,
+            group: currentData.grafana_alert.rule_group,
+            ruleName: currentData.grafana_alert.title,
+          },
+          loading: false,
+        };
+      }
+
+      // In theory, this should never happen
+      return {
+        loading: false,
+        error: new Error(`Unable to obtain rule location for rule ${ruleIdentifier.uid}`),
+      };
+    }
+
+    return {
+      loading: false,
+      error: new Error('Unsupported rule identifier'),
+    };
+  }, [ruleIdentifier, isLoading, isUninitialized, error, currentData]);
+}
+
+function getRulesSourceFromIdentifier(ruleIdentifier: RuleIdentifier): RulesSource | undefined {
+  if (isGrafanaRuleIdentifier(ruleIdentifier)) {
+    return 'grafana';
+  }
+
+  return getDataSourceByName(ruleIdentifier.ruleSourceName);
+}
+
+// This Hook fetches rule definition from the Ruler API only
+export function useRuleWithLocation({
+  ruleIdentifier,
+}: {
+  ruleIdentifier: RuleIdentifier;
+}): RequestState<RuleWithLocation> {
+  const ruleSource = getRulesSourceFromIdentifier(ruleIdentifier);
+
+  const { dsFeatures, isLoadingDsFeatures } = useDataSourceFeatures(ruleIdentifier.ruleSourceName);
```

Comment:
Shouldn't we use the `ruleSource` variable here instead of `ruleIdentifier.ruleSourceName` 🤔